Userland/Services/LookupServer/MulticastDNS.cpp

```diff
@@ -45,8 +46,9 @@ MulticastDNS::MulticastDNS(Object* parent)
         handle_packet();
     };
 
-    // TODO: Announce on startup. We cannot just call announce() here,
-    // because it races with the network interfaces getting configured.
+    m_announce_timer = add<Core::Timer>(10000, [this]() {
+        announce();
+    });
```

Comment:
This doesn't fix the issue I outlined in the comment though; adding a timeout does not guarantee that the network will be up that soon. We need a way to get notifications from the kernel on the IP addresses changing.

Reply:
By the way, the Terrible mDNS responder gets around this by asking SysV init to only run it once "the network is up". Which [is a broken concept](https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/) that we should avoid in Serenity.

Reply:
> adding a timeout does not guarantee that the network will be up that soon.

In particular, [I tend to set up networking manually](https://gist.github.com/bugaevc/027d530554e1657a9fa6e48d0c24075f), which means that the network really becomes "up" only in a minute after boot or so.

Reply:
The timer is recurring though, i.e. it sends that every 10 seconds. This seems to match what other mDNS clients on my network do - maybe with a smaller interval.

Reply:
Oh, that's a bad idea. The RFC explicitly says:

> A Multicast DNS responder MUST NOT send announcements in the absence of information that its network connectivity may have changed in some relevant way.  In particular, a Multicast DNS responder MUST NOT send regular periodic announcements as a matter of course.

Of course, we don't (yet) implement the spec correctly, but it doesn't seem wise to go against it like that. The spec actually wants us to re-announce when our addresses change:

> At any time, if the rdata of any of a host's Multicast DNS records changes, the host MUST repeat the Announcing step described above to update neighboring caches.  For example, if any of a host's IP addresses change, it MUST re-announce those address records.

Which is why I'm saying we need notifications from the kernel.

If we don't want to implement that yet, it should be mostly fine to skip initial announcements. If anybody's interested in our name they'll send a request to which we'll reply. This doesn't guard us against conflicts (which probing is for), or doesn't really work if we want to announce more than just the IP address (for DNS-SD), so we'll have to implement the real thing sooner or later.

Reply:
Aye, that makes sense. I've removed that patch.